src/actions/genSale/purchaseDomains.ts

```diff
@@ -0,0 +1,121 @@
+import * as ethers from "ethers";
+import { getSaleStatus } from ".";
+import { GenSale } from "../../contracts/types";
+import { Claim, SaleStatus, Mintlist, Maybe, GenSaleStatus } from "../../types";
+
+const abi = ["function masterCopy() external view returns (address)"];
+
+const errorCheck = (condition: boolean, errorMessage: string) => {
+    if (condition) {
+        throw new Error(errorMessage);
+    }
+};
+
+const generateAccessList = (
+    userAddress: string,
+    gnosisSafeProxyAddress: string,
+    gnosisSafeImplAddress: string
+) => {
+    return [
```

Comment:
nit: this formatting is 4 spaces (or a tab) and the rest of the code is 2 spaces, please be consistent